libsel4/include/sel4/bootinfo_types.h

```diff
@@ -69,11 +70,24 @@ typedef struct seL4_BootInfo {
     seL4_SlotRegion   schedcontrol; /* Caps to sched_control for each node */
 #endif
     seL4_SlotRegion   untyped;         /* untyped-object caps (untyped caps) */
-    seL4_UntypedDesc  untypedList[CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS]; /* information about each untyped */
-    /* the untypedList should be the last entry in this struct, in order
-     * to make this struct easier to represent in other languages */
+    seL4_UntypedDesc  untypedList[];   /* information about each untyped */
+    /* the untypedList[] must be the last entry */
 } seL4_BootInfo;
 
+/* The constant CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS has been removed from
+ * CMake configuration and a flexible array untypedList[] is used. It counts a
+ * 0 when using sizeof(seL4_BootInfo), so we can define the max number of
+ * elements according to the boot info frame size here. For userland the
+ * specification defines that the boot info frame is 4 KiB, which corresponds to
+ * the page size on x86, ARM and RISC-V, and extra boot info according to
+ * extraLen/extraBIPages follows afterwards.
+ */
+#define BI_FRAME_SIZE_BITS  12  /* 2^12 = 4096 */
+SEL4_COMPILE_ASSERT(seL4_BootInfo_exceeds_4KiB,
+                    sizeof(seL4_BootInfo) <= LIBSEL4_BIT(BI_FRAME_SIZE_BITS));
+#define CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS \
```

Comment:
I would call this something else than CONFIG_* to avoid confusion. True that that's not backward compatible, but I don't think seL4 should strive for backward compatibility when removing unnecessary config options.

Reply:
There is `MAX_NUM_BOOTINFO_UNTYPED_CAPS` now and `CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS`is kept as an alias to avoid breaking existing code.